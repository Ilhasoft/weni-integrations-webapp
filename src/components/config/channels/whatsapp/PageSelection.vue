<template>
  <unnnic-modal
    ref="unnnic-wpp-demo-modal"
    class="wpp-demo-modal"
    text="Selecione sua conta do WhatsApp Business que deseja conectar"
    scheme="feedback-green"
    modal-icon="check-circle-1-1"
    @close="closePopUp"
  >
    <div slot="message">
      <unnnic-button
        v-for="(page, index) in customData.pages"
        :key="index"
        type="secondary"
        size="large"
        :text="page.name"
        @click.stop="selectPage(page)"
      />
    </div>
  </unnnic-modal>
</template>

<script>
  export default {
    name: 'WppDemoModal',
    props: {
      app: {
        type: Object,
        default: /* istanbul ignore next */ () => {},
      },
      customData: {
        type: Object,
        default: () => {},
      },
    },
    data() {
      return {
        showModal: false,
      };
    },
    methods: {
      closePopUp() {
        this.showModal = !this.showModal;
        this.$emit('closePopUp');
      },
      selectPage() {
        this.$router.replace('/apps/my');
      },
    },
  };
</script>

<style lang="scss" scoped></style>
