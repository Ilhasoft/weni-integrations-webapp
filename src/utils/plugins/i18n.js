import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
