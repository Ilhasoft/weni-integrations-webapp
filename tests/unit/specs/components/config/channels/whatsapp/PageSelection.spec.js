import VueRouter from 'vue-router';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import PageSelection from '@/components/config/channels/whatsapp/PageSelection.vue';
import i18n from '@/utils/plugins/i18n';
import { singleApp } from '../../../../../../__mocks__/appMock';

const router = new VueRouter();

const localVue = createLocalVue();
localVue.use(VueRouter);
describe('whatsapp/PageSelection.vue', () => {
  let wrapper;
  beforeEach(() => {
    const customData = {
      pages: [
        { id: 1, name: 'page1' },
        { id: 2, name: 'page2' },
      ],
    };

    wrapper = shallowMount(PageSelection, {
      localVue,
      i18n,
      router,
      propsData: {
        app: singleApp,
        customData,
      },
      mocks: {
        $t: () => 'some specific text',
      },
      stubs: {
        UnnnicModal: true,
        UnnnicButton: true,
        UnnnicSelect: true,
      },
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('closePopUp()', () => {
    it('should set showModal oposite', () => {
      wrapper.setData({ showModal: true });
      expect(wrapper.vm.showModal).toBeTruthy();
      wrapper.vm.closePopUp();
      expect(wrapper.vm.showModal).toBeFalsy();
    });

    it('should call parent closePopUp()', () => {
      expect(wrapper.emitted('closePopUp')).toBeFalsy();
      wrapper.vm.closePopUp();
      expect(wrapper.emitted('closePopUp')).toBeTruthy();
    });
  });

  describe('selectPage()', () => {
    it('should redirect to /apps/my', () => {
      const spy = spyOn(wrapper.vm.$router, 'replace');

      expect(spy).not.toHaveBeenCalled();

      wrapper.vm.selectPage();

      expect(spy).toHaveBeenCalledTimes(1);
      expect(spy).toHaveBeenCalledWith('/apps/my');
    });
  });
});
