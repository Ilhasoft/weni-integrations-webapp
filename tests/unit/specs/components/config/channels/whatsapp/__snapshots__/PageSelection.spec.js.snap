// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`whatsapp/PageSelection.vue should be rendered properly 1`] = `
<unnnicmodal-stub text="some specific text" scheme="feedback-green" modal-icon="phone-3" class="page-selection">
  <div>
    <unnnic-select size="sm" value="[object Object]">
      <option>page.name</option>
      <option>page.name</option>
    </unnnic-select>
  </div>
</unnnicmodal-stub>
`;
